wordpress-mobile/WordPress-Android
# Feature: Search Posts and Pages lists on WP.com/Jetpack sites

Fixes #62! I thought this was my oldest ticket fixed, but that honor belongs to #53. This PR will allow users to search their .com and Jetpack sites posts and pages. It's a long-requested feature so I think a lot of users will enjoy this.

Unfortunately self-hosted sites don't have an endpoint we can use to search remotely. I've left a note in the FluxC implementation to add support for self-hosted sites that have the WordPress REST API plugin available.

Analytics aren't included at the moment. I've opened #5729 to track this.

~A note about this PR: I noticed that the `PostsListAdapter` was handling a lot of work that should be delegated to the Fragment or Activity. I took the liberty of moving FluxC dispatch calls out, but there's still some loading being done that I think we should offload. That's for another PR though, I don't want to overcomplicate this feature.~

To test:
* Sign into a self-hosted site
* Verify that search is not available in the posts and pages lists
* Sign into a Jetpack and/or .com site
* Verify that search is available in the posts and pages lists
* Begin typing and verify that the progress indicator is displayed
* Wait for results (1.5s delay in searching) and verify progress indicator is hidden
* Type more before search results are shown to verify network calls are continuously delayed until user stops typing for 1.5s

cc @oguzkocer, I owe you some reviews by now 😉
